backend/Application/Application.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Application;
+
+use Internships\FileSystem\DirectoryManager;
+use Internships\FileSystem\FileManager;
+use Internships\Services\CompanyDataBuilder;
+use Internships\Services\CsvReader;
+use Internships\Services\FacultyDataBuilder;
+
+class Application
+{
+    protected DirectoryManager $directoryManager;
+    protected FileManager $fileManager;
+    protected CsvReader $csvReader;
+    protected FacultyDataBuilder $facultyBuilder;
+    protected array $childBuilders;
+
+    public function __construct(
+        string $rootPath,
+        string $apiRelativePath,
+        string $resourceRelativePath
+    ) {
+        $this->directoryManager = new DirectoryManager($rootPath, $apiRelativePath, $resourceRelativePath);
+        $this->fileManager = new FileManager($this->directoryManager);
+        $this->csvReader = new CsvReader();
+        $this->facultyBuilder = new FacultyDataBuilder("/faculties/");
+        $this->childBuilders = [
+            new CompanyDataBuilder(""),
+        ];
+    }
+
+    public function build(): void
+    {
+        $jsonPrintFlags = JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT;
+        $facultyCsvData = $this->csvReader->getCSVData(
+            $this->directoryManager->getResourcePath($this->facultyBuilder->getSource()->getRelativePath())
+            . $this->facultyBuilder->getSource()->getFileName()
+        );
+        $faculties = $this->facultyBuilder->buildFromData($facultyCsvData);
```

Comment:
I know it's hard to precisely define, but in place like this an empty line would increase the readability of the file.